### PR TITLE
[pt2] add meta function for `linalg.cross`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #99281
* __->__ #99279
* #99276

### CODE REVIEW

**Comment by nkaretnikov** [comment]
Part of this stack because it unlocks tests close to the ones in the previous PR.

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -88,6 +88,26 @@ def meta_take(self, index, *, out=None):
...
+        self.size(dim) == 3 and other.size(dim) == 3,
+        lambda: (
+            f"linalg.cross: inputs dimension {dim} must have length 3. "
+            f"Got {self.size(dim)} and {other.size(dim)}"
+        ),
+    )
+    out_shape = _broadcast_shapes(self.shape, other.shape)
+    return self.new_empty(out_shape)
+
+
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:110]: See `TORCH_META_FUNC(linalg_cross)` in core.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**